Anytype/Sources/PresentationLayer/Flows/MembersipCoordinator/MembershipCoordinatorModel.swift

```diff
@@ -8,20 +8,32 @@ final class MembershipCoordinatorModel: ObservableObject {
     @Published var showSuccess: MembershipTier?
     @Published var emailVerificationData: EmailVerificationData?
     
+    @Published var userTier: MembershipTier?
+    
+    private let membershipService: MembershipServiceProtocol
+    
     private let membershipAssembly: MembershipModuleAssemblyProtocol
     private let tierSelectionAssembly: MembershipTierSelectionAssemblyProtocol
     private let emailVerificationAssembly: EmailVerificationAssemblyProtocol
     
     init(
+        membershipService: MembershipServiceProtocol,
         membershipAssembly: MembershipModuleAssemblyProtocol,
         tierSelectionAssembly: MembershipTierSelectionAssemblyProtocol,
         emailVerificationAssembly: EmailVerificationAssemblyProtocol
     ) {
+        self.membershipService = membershipService
         self.membershipAssembly = membershipAssembly
         self.tierSelectionAssembly = tierSelectionAssembly
         self.emailVerificationAssembly = emailVerificationAssembly
     }
     
+    func onAppear() {
```

Comment:
I think you can just mark onAppear as async here

Reply:
membershipService.getStatus() throws
but View modificator .task {} is non throwable
So I have to do catch in view body or wrap call in Task here

Reply:
got it!